### ISSUE TITLE
`Style/SuperWithArgsParentheses` and `Style/MethodCallWithArgsParentheses` cause infinite loop

### ISSUE THREAD

**Issue by modosc** [rubocop/rubocop#12437]
## Expected behavior

i'm unsure what the behavior should be here but these two rules are in conflict. I think the best case scenario is that there's no infinite loop / long error backtrace dump

## Steps to reproduce the problem

```ruby
# frozen_string_literal: true

class Foo
  def foo(arg)
    pp arg
  end
end

class Bar < Foo
  def foo(arg)
   super(arg)
  end
end
```

```
$ rubocop --only Style/SuperWithArgsParentheses,Style/MethodCallWithArgsParentheses -A foo.rb Inspecting 1 file
C

Offenses:

foo.rb:11:4: C: [Corrected] Style/SuperWithArgsParentheses: Use parentheses for super with arguments.
   super arg
   ^^^^^^^^^
foo.rb:11:9: C: [Corrected] Style/MethodCallWithArgsParentheses: Omit parentheses for method calls with arguments.
   super(arg)
        ^^^^^

0 files inspected, 2 offenses detected, 2 offenses corrected
Infinite loop detected in /Users/jon/git/horizon-api/foo.rb and caused by Style/MethodCallWithArgsParentheses -> Style/SuperWithArgsParentheses
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/runner.rb:332:in `check_for_infinite_loop'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/runner.rb:315:in `block in iterate_until_no_changes'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/runner.rb:314:in `loop'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/runner.rb:314:in `iterate_until_no_changes'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/runner.rb:283:in `do_inspection_loop'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/runner.rb:164:in `block in file_offenses'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/runner.rb:189:in `file_offense_cache'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/runner.rb:163:in `file_offenses'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/runner.rb:154:in `process_file'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/runner.rb:135:in `block in each_inspected_file'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/runner.rb:134:in `each'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/runner.rb:134:in `reduce'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/runner.rb:134:in `each_inspected_file'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/runner.rb:120:in `inspect_files'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/runner.rb:73:in `run'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/cli/command.rb:11:in `run'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/cli/environment.rb:18:in `run'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/cli.rb:118:in `run_command'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/cli.rb:125:in `execute_runners'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/cli.rb:51:in `block in run'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/cli.rb:77:in `profile_if_needed'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/lib/rubocop/cli.rb:43:in `run'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/exe/rubocop:19:in `block in <top (required)>'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/benchmark-0.3.0/lib/benchmark.rb:313:in `realtime'
/Users/jon/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/exe/rubocop:19:in `<top (required)>'
/Users/jon/.rbenv/versions/3.2.2/bin/rubocop:25:in `load'
/Users/jon/.rbenv/versions/3.2.2/bin/rubocop:25:in `<main>'
```

## RuboCop version

```
$ rubocop -V
1.58.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 3.2.2) [arm64-darwin22]
  - rubocop-factory_bot 2.24.0
  - rubocop-performance 1.19.1
  - rubocop-rails 2.22.2
  - rubocop-rspec 2.25.0
  ```

**Comment by KessaPassa** [comment]
I cloud not reproduce this.
any other required condition?
there looks to be  difference about system architecture  `arm64-darwin22` vs `x86_64-darwin22` .

```
% bundle exec exe/rubocop --only Style/SuperWithArgsParentheses,Style/MethodCallWithArgsParentheses -A foo.rb
Inspecting 1 file
C

Offenses:

foo.rb:5:5: C: [Corrected] Style/MethodCallWithArgsParentheses: Use parentheses for method calls with arguments.
    pp arg
    ^^^^^^

1 file inspected, 1 offense detected, 1 offense corrected
```

# Rubocop versions
```
1.58.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 3.2.2) [x86_64-darwin22]
  - rubocop-factory_bot 2.24.0
  - rubocop-performance 1.19.1
  - rubocop-rails 2.22.2
  - rubocop-rspec 2.25.0
```

**Comment by modosc** [comment]
ah, forgot we have a configuration for `Style/MethodCallWithAgsParentheses`:

```
Style/MethodCallWithArgsParentheses:
  EnforcedStyle: omit_parentheses
  AllowParenthesesInMultilineCall: true
```

so that's why it's triggered locally.



i'm not sure if there's prior art for other rules which are similar but result in conflicts like this. i guess the ask would be to print some warning if a conflicting config is detected rather than erroring out? happy to just close this out otherwise.

**Comment by KessaPassa** [comment]
thank you!
I was able to reproduce infinite loop.

**Comment by KessaPassa** [comment]
[Ruby Style Guide](https://rubystyle.guide/#introduction) says
> RuboCop is a static code analyzer (linter) and formatter, based on this style guide.

Looking at the [guide of super method](https://rubystyle.guide/#super-with-args), it seems that `super(args)` is correct.
In my opinion, it is correct like this code.
```rb
# frozen_string_literal: true

class Foo
  def foo(arg)
    pp arg
  end
end

class Bar < Foo
  def foo(arg)
    super(arg)
  end
end
```

However, the problem is that an infinite loop occurs.
In my opinion, It should be modified to avoid an infinite loop when this command executed.
```
bundle exec exe/rubocop --only Style/SuperWithArgsParentheses,Style/MethodCallWithArgsParentheses -A foo.rb
```

ref: https://github.com/rubocop/rubocop/pull/12390
